common/ledger/blkstorage/blockindex.go

```diff
@@ -223,6 +223,24 @@ func (index *blockIndex) getTxValidationCodeByTxID(txID string) (peer.TxValidati
 	return peer.TxValidationCode(v.TxValidationCode), nil
 }
 
+func (index *blockIndex) txIDExists(txID string) (bool, error) {
+	if !index.isAttributeIndexed(IndexableAttrTxID) {
+		return false, ErrAttrNotIndexed
+	}
+	rangeScan := constructTxIDRangeScan(txID)
+	itr, err := index.db.GetIterator(rangeScan.startKey, rangeScan.stopKey)
+	if err != nil {
+		return false, errors.WithMessagef(err, "error while trying to check the presence of TXID [%s]", txID)
+	}
+	defer itr.Release()
+
+	present := itr.Next()
```

Comment:
so, just to understand how we're changing this: did we use to return `ledger.NotFoundInIndexErr` before if present was false?

Reply:
We are not changing the existing function but yes in that function it returns an error at this point. The existing function `GetTransactionByID(txID string) (*peer.ProcessedTransaction, error)` is more comprehensive that was originally meant for a general purpose client that wants to retrieve a `TxEnvelope`. However when it was used in the validation path, it became necessary to distinguish the error types (because the fist return would always be nil in that case). 
For your amusement, you has fixed this in the validation path - https://jira.hyperledger.org/browse/FAB-10201 :-)